### DISCUSSION TITLE
postUpgradeTasks actions

### DISCUSSION THREAD

**Discussion by jonyscathe** [renovatebot/renovate#21078] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

Hi,

I am currently using postUpgradeTasks to run various linters/formatters that have just been updated.
For example, for when the pre-commit hook for black updates I have:
```
  "packageRules": [
      "matchPackageNames": [
        "psf/black"
      ],
      "postUpgradeTasks": {
        "commands": [
          "pre-commit run --all-files black || pre-commit run --all-files black"
      ],
      "fileFilters": [
        "**/*.py"
      ]
  ]
```
This works really well. Unfortunately pre-commit has to be run twice (first run will fail if it modifies files, second run should almost always pass) but other than that, works really well.

The only thing that I would love to do differently is have the postUpgradeTasks stop an automerge if (and only if) the postUpgradeTasks actually modifies or creates a file.

The logic here being that if something like black updates in 30 of my projects at once.
If in 25 of those, black makes zero updates, then automerging those projects is great.
But if black makes changes in the other 5 projects, then actually reviewing those projects would be useful in case black has broken something.

As far as I can tell that isn't possible

The same thing could be achieved if it was possible to have automerge set to false if any files have been updated that match a certain pattern (in the case above if anything matching `*.py` has changed). But I can't see any way to do that.

**Comment by rarkins** [comment]
This isn't possible. I think it could be done as a github action or CI check though, independently of Renovate
